core/capabilities/ccip/ccipsolana/executecodec.go

```diff
@@ -0,0 +1,258 @@
+	if err != nil {
+		return cciptypes.ExecutePluginReport{}, fmt.Errorf("unpack encoded report: %w", err)
+	}
+
+	tokenAmounts := make([]cciptypes.RampTokenAmount, 0, len(executeReport.Message.TokenAmounts))
+	for _, tokenAmount := range executeReport.Message.TokenAmounts {
+		destData := make([]byte, 4)
+		binary.LittleEndian.PutUint32(destData, tokenAmount.DestGasAmount)
+
+		tokenAmounts = append(tokenAmounts, cciptypes.RampTokenAmount{
+			SourcePoolAddress: tokenAmount.SourcePoolAddress,
+			DestTokenAddress:  tokenAmount.DestTokenAddress.Bytes(),
+			ExtraData:         tokenAmount.ExtraData,
+			Amount:            decodeLEToBigInt(tokenAmount.Amount.LeBytes[:]),
+			DestExecData:      destData,
+		})
+	}
+
+	var buf bytes.Buffer
+	encoder := agbinary.NewBorshEncoder(&buf)
+	err = executeReport.Message.ExtraArgs.MarshalWithEncoder(encoder)
+	if err != nil {
+		return cciptypes.ExecutePluginReport{}, fmt.Errorf("unpack encoded report: %w", err)
+	}
+
+	messages := []cciptypes.Message{
+		{
+			Header: cciptypes.RampMessageHeader{
+				MessageID:           executeReport.Message.Header.MessageId,
+				SourceChainSelector: cciptypes.ChainSelector(executeReport.Message.Header.SourceChainSelector),
+				DestChainSelector:   cciptypes.ChainSelector(executeReport.Message.Header.DestChainSelector),
+				SequenceNumber:      cciptypes.SeqNum(executeReport.Message.Header.SequenceNumber),
+				Nonce:               executeReport.Message.Header.Nonce,
+				MsgHash:             cciptypes.Bytes32{},        // todo: info not available, but not required atm
+				OnRamp:              cciptypes.UnknownAddress{}, // todo: info not available, but not required atm
+			},
+			Sender:         executeReport.Message.Sender,
+			Data:           executeReport.Message.Data,
+			Receiver:       executeReport.Message.LogicReceiver.Bytes(),
+			ExtraArgs:      buf.Bytes(),
+			FeeToken:       cciptypes.UnknownAddress{}, // <-- todo: info not available, but not required atm
+			FeeTokenAmount: cciptypes.BigInt{},         // <-- todo: info not available, but not required atm
+			TokenAmounts:   tokenAmounts,
+		},
+	}
+
+	offchainTokenData := make([][][]byte, 0, 1)
+	if executeReport.OffchainTokenData != nil {
+		offchainTokenData = append(offchainTokenData, executeReport.OffchainTokenData)
+	}
+
+	proofs := make([]cciptypes.Bytes32, 0, len(executeReport.Proofs))
+	for _, proof := range executeReport.Proofs {
+		proofs = append(proofs, proof)
+	}
+
+	chainReport := cciptypes.ExecutePluginReportSingleChain{
+		SourceChainSelector: cciptypes.ChainSelector(executeReport.SourceChainSelector),
+		Messages:            messages,
+		OffchainTokenData:   offchainTokenData,
+		Proofs:              proofs,
+	}
+
+	report := cciptypes.ExecutePluginReport{
+		ChainReports: []cciptypes.ExecutePluginReportSingleChain{chainReport},
+	}
+
+	return report, nil
+}
+
+func parseExtraArgsMap(input map[string]any) (ccip_router.SVMExtraArgs, error) {
```

Comment:
Maybe another ticket/PR, but we would ideally also like to do the same on the EVM codec eventually.